src/graph/LookupExecutor.cpp

```diff
@@ -0,0 +1,499 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "graph/LookupExecutor.h"
+
+namespace nebula {
+namespace graph {
+
+LookupExecutor::LookupExecutor(Sentence *sentence, ExecutionContext *ectx)
+    : TraverseExecutor(ectx, "lookup") {
+    sentence_ = static_cast<LookupSentence*>(sentence);
+}
+
+Status LookupExecutor::prepare() {
+    return Status::OK();
+}
+
+Status LookupExecutor::prepareClauses() {
+    DCHECK(sentence_ != nullptr);
+    Status status = Status::OK();
+    spaceId_ = ectx()->rctx()->session()->space();
+    expCtx_ = std::make_unique<ExpressionContext>();
+    expCtx_->setStorageClient(ectx()->getStorageClient());
+    expCtx_->setSpace(spaceId_);
+    do {
+        status = checkIfGraphSpaceChosen();
+        if (!status.ok()) {
+            break;
+        }
+        status = prepareFrom();
+        if (!status.ok()) {
+            break;
+        }
+        status = prepareWhere();
+        if (!status.ok()) {
+            break;
+        }
+        status = prepareYield();
+        if (!status.ok()) {
+            break;
+        }
+    } while (false);
+
+    if (!status.ok()) {
+        LOG(ERROR) << "Preparing failed: " << status;
+        return status;
+    }
+    return status;
+}
+
+Status LookupExecutor::prepareFrom() {
+    from_ = sentence_->from();
+    auto *mc = ectx()->getMetaClient();
+    auto tagResult = mc->getTagIDByNameFromCache(spaceId_, *from_);
+    if (tagResult.ok()) {
+        tagOrEdge_ = tagResult.value();
+        return Status::OK();
+    }
+    auto edgeResult = mc->getEdgeTypeByNameFromCache(spaceId_, *from_);
+    if (edgeResult.ok()) {
+        tagOrEdge_ = edgeResult.value();
+        isEdge_ = true;
+        return Status::OK();
+    }
+    LOG(ERROR) << "Tag or Edge was not found : " << from_->c_str();
+    return Status::Error("Tag or Edge was not found : %s", from_->c_str());
+}
+
+Status LookupExecutor::prepareWhere() {
+    auto *clause = sentence_->whereClause();
+    whereWrapper_ = std::make_unique<WhereWrapper>(clause);
```

Comment:
We don't need WhereWrapper here,  all filters should be pushed down to the storaged.